src/actions/events.js

```diff
@@ -186,6 +191,91 @@ events.on(GET_BUILD_LIST, function(event) {
       builds.map(function(build) {
         tree.set(['builds', owner, name, build.number], build);
       });
+
+      // populate filter suggestions
+      (function() {
+        let terms = Object.keys(builds).map(k => builds[k]).reduce((dict, build) => {
+          dict['author:' + build.author] = true;
+          dict['branch:' + build.branch] = true;
+          dict['status:' + build.status] = true;
+          dict['event:' + build.event] = true;
+          if(build.ref.includes('tags')) {
+            dict['tag:' + build.ref.replace('refs/tags/', '')] = true;
+          }
+          if(build.deploy_to !== '') {
+            dict['deploy_to:' + build.deploy_to] = true;
+          }
+          return dict;
+        }, {});
+
+        let suggestions = Object.keys(terms)
+          .sort((a, b) => a > b)
+          .map(item => ({label:item, value:item}));
+        events.emit(BUILD_FILTER_SUGGESTIONS, {owner, name, suggestions});
```

Comment:
This has been ported from the component so perhaps we could just set the value in the tree in here instead of emitting another event to do that? Not sure if triggering event from another event's handler is an anti-pattern here.